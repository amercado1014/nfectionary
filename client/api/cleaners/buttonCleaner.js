export const buttonCleaner = (names, count) => {
  const buttonData = names.reduce((accu, name) => {
    count.filter(num => {
      let newObject = {};
      if (num.id === name.id) {
        newObject['name'] = name.name;
        newObject['count'] = num.count;
        newObject['disease_id'] = num.id;
      }
      if (newObject.name) {
        return accu.push(newObject);
      }
    });
    return accu;
  }, []);
  return buttonData;
};
