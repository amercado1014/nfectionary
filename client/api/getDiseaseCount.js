export const getDiseaseCount = async (id) => {
  try {
    const response = await fetch(`http://localhost:5000/api/v1/state-diseases/${id}`);
    const diseases = await response.json();
    const diseaseCount = diseases.map(disease => ({ 
      count: disease.case_count, 
      id: disease.diseases_id 
    }));
    return diseaseCount
  } catch(err) {
    throw new Error(`Error fetching disease case count: ${err.message}`);
  }
};