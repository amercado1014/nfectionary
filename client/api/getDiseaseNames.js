export const getDiseaseNames = async () => {
  const url = 'https://nfectionary.herokuapp.com/api/v1/diseases';
  try {
    const response = await global.fetch(url);
    const diseases = await response.json();
    
    const diseaseNames = diseases.map(disease => ({
      name: disease.name,
      id: disease.id
    }));
    return diseaseNames;
  } catch (err) {
    throw new Error(`Error fetching disease names: ${err.message}`);
  }
};