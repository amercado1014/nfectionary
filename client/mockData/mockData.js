export const diseaseNames = [
  {
    'id': 1,
    "name": "Influenza"
  },
  {
    "id": 2,
    "name": "Salmonella"
  }
]

export const diseaseArrayInfo = [
  {
    'id': 1,
    "name": "Influenza",
    "treatment":
      "Bedrest and plenty of fluids. In some cases, doctors may prescribe an antiviral medication, such as Oseltamivir or Zanamivir.",
    "signs_symptoms":
      "People who have the flu often feel some or all of these symptoms: fever, cough, sore throat, runny, stuffy nose, muscle, body aches, headaches, fatigue, vomiting, diarrhea",
    "preventative_measures":
      "The Centers for Disease Control and Prevention recommends annual flu vaccination for everyone over the age of 6 months. Wash your hands, contain your coughs, avoid crowds.",
    "testing_procedures":
      "rapid influenza diagnostic tests which tests for the parts of the virus (antigen). Molecular testing, DNA PCR testing.",
    "images":
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR3W0jSwSscVIFRF7qDo0eW28bvPR6AGu0-t94TmBcvfbM4OblM-A",
    "transmission":
      "Flu viruses travel through the air in droplets when someone with the infection coughs, sneezes or talks. You can inhale the droplets directly, or you can pick up the germs from an object — such as a telephone or computer keyboard — and then transfer them to your eyes, nose or mouth.",
    "summary":
      "Influenza (flu) is a contagious respiratory illness caused by influenza viruses. It can cause mild to severe illness. Serious outcomes of flu infection can result in hospitalization or death. Some people, such as older people, young children, and people with certain health conditions, are at high risk of serious flu complications."
  },
  {
    "id": 2,
    "name": "Salmonella",
    "treatment":
      "Antibiotics. If your doctor suspects that salmonella bacteria have entered your bloodstream, or if you have a severe case or a compromised immune system, he or she may prescribe antibiotics to kill the bacteria. Antibiotics are not of benefit in uncomplicated cases. In fact, antibiotics may prolong the period in which you carry the bacteria and can infect others, and they can increase your risk of relapse.",
    "signs_symptoms":
      "nausea, vomiting, abdominal cramps, diarrhea, fever, chills, headaches, blood in stool",
    "preventative_measures":
      "Cook poultry, ground beef and eggs thoroughly. Do not drink or eat raw eggs or milk. Wash hands, kitchen surfaces and utensils with soap and water immediately after contact with raw meat or poultry. Wash hands after handling reptiles, birds, and pet feces.",
    "testing_procedures":
      "Culture: Laboratory scientists identify Salmonella infection by culturing a patient’s sample. If Salmonella bacteria grow, then the diagnosis is confirmed",
    "images":
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSdN45BCFJ0Bk5tKJEucvvL25P8hsX25RKDthezUxqCXvC7MdpR",
    "transmission":
      "Eating food contaminated by feces such as meats, poultry, raw eggs, fruits, vegetables or seafood.",
    "summary":
      "Salmonella infection (salmonellosis) is a common bacterial disease that affects the intestinal tract. Salmonella bacteria typically live in animal and human intestines and are shed through feces. Humans become infected most frequently through contaminated water or food."
  }
]

export const diseaseCount = [
  {
    id: 1,
    diseases_id: 1,
    states_id: 1,
    case_count: 40
  }, 
  {
    id: 2,
    diseases_id: 2,
    states_id: 1,
    case_count: 20
  }
]

export const returnedDiseaseCount = [
  {
    id: 1,
    count: 40
  }, 
  {
    id: 2,
    count: 20
  }
]

export const returnedGraphCounts = [
  {
    count: 40,
    id: 1,
    state: 1
  },
  {
    count: 20,
    id: 1,
    state: 2
  }
]

export const diseaseObject =   {
  "name": "Influenza",
  "treatment":
    "Bedrest and plenty of fluids. In some cases, doctors may prescribe an antiviral medication, such as Oseltamivir or Zanamivir.",
  "signs_symptoms":
    "People who have the flu often feel some or all of these symptoms: fever, cough, sore throat, runny, stuffy nose, muscle, body aches, headaches, fatigue, vomiting, diarrhea",
  "preventative_measures":
    "The Centers for Disease Control and Prevention recommends annual flu vaccination for everyone over the age of 6 months. Wash your hands, contain your coughs, avoid crowds.",
  "testing_procedures":
    "rapid influenza diagnostic tests which tests for the parts of the virus (antigen). Molecular testing, DNA PCR testing.",
  "images":
    "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR3W0jSwSscVIFRF7qDo0eW28bvPR6AGu0-t94TmBcvfbM4OblM-A",
  "transmission":
    "Flu viruses travel through the air in droplets when someone with the infection coughs, sneezes or talks. You can inhale the droplets directly, or you can pick up the germs from an object — such as a telephone or computer keyboard — and then transfer them to your eyes, nose or mouth.",
  "summary":
    "Influenza (flu) is a contagious respiratory illness caused by influenza viruses. It can cause mild to severe illness. Serious outcomes of flu infection can result in hospitalization or death. Some people, such as older people, young children, and people with certain health conditions, are at high risk of serious flu complications."
  }

export const stateArray = [
  {
    "id": 1,
    "name": "Alabama"
  },
  {
    "id": 2,
    "name": "Alaska"
  },
  {
    "id": 3,
    "name": "Arizona"
  },
  {
    "id": 4,
    "name": "Arkansas"
  }
]

export const stateNames = ["Alabama", "Alaska", "Arizona", "Arkansas"]

export const returnedButtonData = [{ 
  name: 'Influenza', 
  count: 40, 
  disease_id: 1 
}, 
{ 
  name: 'Salmonella', 
  count: 20, 
  disease_id: 2 
}
];

export const stringId = '1';

export const diseases = ['Influenza', 'Salmonella']

export const returnedGraphData = [
  {
    state: 'AL',
    count: 40
  },
  {
    state: 'AK',
    count: 20
  }
]