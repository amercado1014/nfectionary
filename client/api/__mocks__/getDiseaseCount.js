import * as mock from '../../mockData/mockData';

export const getDiseaseCount = jest.fn().mockImplementation(() => {
  return Promise.resolve(mock.diseaseCount);
});
