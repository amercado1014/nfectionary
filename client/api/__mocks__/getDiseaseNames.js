import * as mock from '../../mockData/mockData';

export const getDiseaseNames = jest.fn().mockImplementation(()=> {
  return Promise.resolve(mock.diseaseArrayInfo)
})