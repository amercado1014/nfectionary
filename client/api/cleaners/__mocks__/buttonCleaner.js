import * as mock from '../../../mockData/mockData';

export const buttonCleaner = jest.fn().mockImplementation(() => {
  return mock.returnedButtonData;
});
