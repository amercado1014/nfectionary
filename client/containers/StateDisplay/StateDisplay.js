import React, { Component } from 'react';
import { 
  StyleSheet, 
  Text, 
  View, 
  TouchableOpacity
} from 'react-native';
import ModalDropdown from 'react-native-modal-dropdown';
import { getDiseaseNames } from '../../api/getDiseaseNames';
import { getDiseaseCount } from '../../api/getDiseaseCount';
import { buttonCleaner } from '../../api/cleaners/buttonCleaner';
import { diseaseSort } from '../../api/cleaners/sortDiseaseData';
import PropTypes from 'prop-types';
export default class StateDisplay extends Component {
  constructor(props) {
    super(props);
    this.state = {
      buttonData: [],
      state: '',
      statesList: [],
      diseaseList: []
    };
  }

  async componentDidMount() {
    const id = this.props.navigation.getParam('id');
    await this.fetchAllData(id);
  }

  async fetchAllData(id) {
    const diseaseId = Number(id) + 1;
    const diseaseList = await getDiseaseNames();

    diseaseList.sort(diseaseSort);
    const count = await getDiseaseCount(diseaseId);
    const buttonData = buttonCleaner(diseaseList, count);
    
    this.setButtonData(buttonData, id, diseaseList);
  }

  setButtonData(buttonData, id, diseaseList) {
    const statesList = this.props.navigation.getParam('statesList');
    const state = statesList[id];
    
    this.setState({ buttonData, state, statesList, diseaseList });
  }

  static navigationOptions = {
    title: ' ',
    headerStyle: {
      backgroundColor: '#3E79CA',
      borderBottomColor: '#3E79CA'
    },
    headerTintColor: '#fff',
    headerTitleStyle: {
      color: '#ffffff',
      fontSize: 30,
      textDecorationLine: 'underline'
    }
  };
  
  render() {
    const { buttonData, state, statesList, diseaseList } = this.state;

    const renderButton = buttonData.map((button, index) => (
      <TouchableOpacity
        style={styles.button}
        key={button.count + index}
        onPress={() => this.props.navigation.navigate('DiseaseDisplay', 
          {disease_id: button.disease_id, diseaseList })}>
        <Text style={styles.buttonName}> {button.name} </Text>
        <Text style={styles.buttonCount}> {button.count} </Text>
      </TouchableOpacity>
    ));
    
    return (
      <View style={styles.container}>
        <View style = {styles.dropDownWrap}>
          <ModalDropdown 
            defaultValue={state}
            style={styles.dropDown}
            options={statesList} 
            textStyle={{ color: '#3E79CA', fontSize: 20, textAlign: 'left', width: 160 }}
            dropdownTextStyle={ {color: '#3E79CA', fontSize: 16} }
            onSelect={(event) => this.fetchAllData(event)} 
          />
          <View style={styles.arrWrap}><Text style={styles.arrColor}>â–¼</Text></View>
        </View>
      
        <View style={styles.buttonWrap}>
          {renderButton}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#3E79CA',
    alignItems: 'center',
    justifyContent: 'center'
  },
  dropDownWrap: {
    alignItems:'center', 
    flexDirection:'row'
  },
  dropDown: {
    backgroundColor: '#ffffff',
    marginBottom: 20,
    padding: 10,
    width: 190,
    borderRadius: 3,
    shadowOffset:{  width: 1,  height: 1,  },
    shadowColor: 'black',
    shadowOpacity: .5
  },
  arrWrap: {
    position: "absolute", 
    right: 15, 
    top: 15
  },
  arrColor: {
    color: '#3E79CA'
  },
  buttonWrap: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
  },
  button: {
    backgroundColor: 'rgba(255,255,255,.3)',
    margin: 10,
    width: 160,
    height: 75,
    padding: 10,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
    borderWidth: 1,
    borderColor: '#ffffff',
    shadowOffset:{  width: 1,  height: 1,  },
    shadowColor: 'black',
    shadowOpacity: .5
  },
  buttonName: {
    fontSize: 20,
    color: '#ffffff'
  },
  buttonCount: {
    fontSize: 18,
    color: '#225293',
    alignItems: 'center',
  }
});

StateDisplay.propTypes = {
  navigation: PropTypes.object
};