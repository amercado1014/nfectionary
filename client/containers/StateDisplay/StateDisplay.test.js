import React from 'react';
import StateDisplay from './StateDisplay';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import { getDiseaseNames } from '../../api/getDiseaseNames';
import { getDiseaseCount } from '../../api/getDiseaseCount';
import { buttonCleaner } from '../../api/cleaners/buttonCleaner';
import * as mock from '../../mockData/mockData';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../api/getDiseaseNames');
jest.mock('../../api/getDiseaseCount');
jest.mock('../../api/cleaners/buttonCleaner');

describe ('StateDisplay', () => {
  let wrapper;
  let instance;
  let buttonData;
  let diseaseList;
  let id;
  let count;

  beforeEach(() => {
    wrapper = shallow(<StateDisplay navigation={{ getParam: jest.fn().mockImplementation(()=> {
      return mock.stateNames }) 
    }} />);
    instance = wrapper.instance();
    buttonData = mock.returnedButtonData;
    diseaseList = mock.diseaseNames;
    id = '1';
    count = mock.returnedDiseaseCount;
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchAllData on componentDidMount', async () => {
    instance.fetchAllData = jest.fn();
    await instance.componentDidMount();
    expect(instance.fetchAllData).toHaveBeenCalled();
  });

  it('should call getDiseaseNames on fetchAllData', async () => {
    await instance.fetchAllData(id);
    expect(getDiseaseNames).toHaveBeenCalled();
  });

  it('should call getDiseaseCount on fetchAllData with correct params', async () => {
    const expected = 2;
    await instance.fetchAllData(id);
    expect(getDiseaseCount).toHaveBeenCalledWith(expected);
  });

  it('should call buttonCleaner on fetchAllData with correct params', async () => {
    await instance.fetchAllData(id);
    expect(buttonCleaner).toHaveBeenCalledWith(diseaseList, count);
  });

  it('should call setButtonData on fetchAllData with correct params', async () => {
    instance.setButtonData = jest.fn();
    await instance.fetchAllData(id);
    expect(instance.setButtonData).toHaveBeenCalledWith(buttonData, id, diseaseList);
  });

  it('should set state with buttonData on setButtonData', () => {
    instance.setButtonData(buttonData, id, diseaseList);
    expect(wrapper.state('buttonData')).toEqual(buttonData);
  });

  it('should set state with the state name on setButtonData', () => {
    const expected = 'Alaska';
    instance.setButtonData(buttonData, id, diseaseList);
    expect(wrapper.state('state')).toEqual(expected);
  });

  it('should set state with statesList on setButtonData', () => {
    const expected = mock.stateNames;
    instance.setButtonData(buttonData, id, diseaseList);
    expect(wrapper.state('statesList')).toEqual(expected);
  });

  it('should set state with diseaseList on setButtonData', () => {
    instance.setButtonData(buttonData, id, diseaseList);
    expect(wrapper.state('diseaseList')).toEqual(diseaseList);
  });
});
