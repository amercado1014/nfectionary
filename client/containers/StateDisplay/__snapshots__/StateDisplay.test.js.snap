// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateDisplay renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    animated={true}
    defaultIndex={-1}
    defaultValue=""
    disabled={false}
    keyboardShouldPersistTaps="never"
    onSelect={[Function]}
    options={Array []}
    scrollEnabled={true}
    showsVerticalScrollIndicator={true}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 12,
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
